TempleEight/docs/guide/enumeration.md

```diff
@@ -3,3 +3,214 @@ id: enumeration
+
+## A basic list endpoint
+In the [Getting Started](../getting-started) guide, we defined the following Templefile:
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+To add an additional `list` endpoint to our `ExampleService`, we can simply add the `#enumerable` metadata tag to that service:
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+
+  #enumerable;
+}
+```
+
+This generates a new endpoint, accessible at `/api/example-service/all`, and this will return all stored elements as a JSON list.
+Let's make some example requests to show it in action:
+
+```bash
+# Deploy the service locally
+❯❯❯ source deploy.sh
+...
+
+# Create two new entities in that service
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "Hello", "bar": 123}'
+{"id":"bb3bdc7f-857e-11ea-bd0b-0242ac130002","foo":"Hello","bar":123}
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "World", "bar": 456}'
+{"id":"cc2e9d07-857e-11ea-bd0b-0242ac130002","foo":"World","bar":456}
+
+# Read back both entries using the list endpoint
+curl -X GET $KONG_ENTRY/api/example-service/all
+{ 
+  "ExampleServiceList":[
+    {"id":"bb3bdc7f-857e-11ea-bd0b-0242ac130002","foo":"Hello","bar":123},
+    {"id":"cc2e9d07-857e-11ea-bd0b-0242ac130002","foo":"World","bar":456}
+  ]
+}
+
+```
+
+## Enumerating services when using authentication
+Authenticating users is one of Temple's core offerings and is discussed in detail in the [authentication](authentication) guide.
+Here, we'll assume you understand what it means to be using the `#authMethod` and `#auth` metadata in your project and services.
+
+One advantage of using authentication is that we can restrict the response of the list endpoint to only include relevant information to the authentication token provided.
+This means that we can restrict responses to only include entities that the authentication token has created.
+
+For example, let's add authentication to our `ExampleProject`, by including a `User` service, and using `email` as the `#authMethod`:
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #authMethod(email);
+}
+
+ExampleUser: service {
+  name: string;
+
+  #auth;
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+
+  #enumerable;
+}
+```
+
+Now when making requests to the `ExampleService` we'll need to include an authorization header.
+Let's show an example of listing:
+
+```bash
+# Deploy the project locally
+❯❯❯ source deploy.sh
+...
+
+# Create two access tokens, extract them from the JSON response, and assign each to an environment variable
+❯❯❯ TOKEN1=$(curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "temple1@temple.com", "password":"abcdefgh"}' | jq -r .AccessToken)
```

Comment:
Is installation of `jq` covered elsewhere? I don’t have it

Reply:
Ah good point, I'll mention it above this, as this is the first time it's used